okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpClient.java

```diff
@@ -64,4 +67,11 @@ RemotingOkHttpCall newCallWithMutableState(
                 executionExecutor,
                 maxNumRelocations);
     }
+
+    private Request addRateLimitIdTag(Request request) {
```

Comment:
we can only mutate request when creating so we had to move acquire to this level. This might have unintended consequence of acquiring limits too early in retrofit clients. We can put a mutable value on the tag so we can set it later but not sure if that's better

Reply:
In that case, let's put a SettableFuture here and initialize it in `enqueue` as discussed offline

Reply:
settable future it is